core/services/ocr2/plugins/functions/plugin.go

```diff
@@ -65,6 +65,130 @@ func NewFunctionsServices(functionsOracleArgs, thresholdOracleArgs, s4OracleArgs
 		return nil, err
 	}
 
+	allServices := []job.ServiceCtx{}
+	var err error
+	switch pluginConfig.ContractVersion {
+	case 0:
```

Comment:
Do we need two versions here at all? Isn't only Listener (or some config option for Listener) enough?

Reply:
Gateway's allowlist also, but I think I can get rid of this switch statement by using a common contract interace

Reply:
Actually gateway's allowlist isn't initialized here